include/boost/beast/core/detail/bind_handler.hpp

```diff
@@ -153,6 +154,14 @@ class bound_handler
         return asio_handler_is_continuation(std::addressof(h->h_));
     }
 
+    template<class Function>
+    friend
+    void asio_handler_invoke(Function&& f, bound_handler* h)
+    {
+        using boost::asio::asio_handler_invoke;
+        asio_handler_invoke(f, std::addressof(h->h_));
```

Comment:
Shouldn't perfect forwarding be applied to `f`?

Reply:
I used `Function&&` to combine `Function const&` and `Function&` into a single overload, `asio_handler_invoke` does not take ownership. Do I still need to use perfect forwarding?

Reply:
You can just make it `Function&` since `Function` is a template parameter. It will be appropriate for both `const U&` and `U&` for any type `U`.

Reply:
But what happens if it is invoked with a lambda? I get a compile error, "A non-const reference may only be bound to an lvalue"

Reply:
Yes. If that is something you want to allow here, then `Function&&` is fine (or two overloads, `const Function&`, and for `Function&`).

Reply:
In other words, it is correct as written :) :) :)

Reply:
> I used Function&& to handle Function const& and Function& in a single overload, asio_handler_invoke does not take ownership. Do I still need to use perfect forwarding?

Thanks for clarifying. My bad

Reply:
Thanks for looking!